homeassistant/components/sensor/dsmr.py

```diff
@@ -0,0 +1,153 @@
+"""
+Support for Dutch Smart Meter Requirements.
+
+Also known as: Smartmeter or P1 port.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.dsmr/
+"""
+
+import asyncio
+import logging
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_DEVICE
+from homeassistant.helpers.entity import Entity
+
+DOMAIN = 'dsmr'
+
+REQUIREMENTS = ['dsmr-parser==0.2']
+
+CONF_DSMR_VERSION = 'dsmr_version'
+DEFAULT_DEVICE = '/dev/ttyUSB0'
+DEFAULT_DSMR_VERSION = '4'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
```

Comment:
No, it's not documented. I'm trying to establish a common base for the configuration and not use `usb_path`, `filename`, and others. It's a connection over a serial port. Even better than `port` would `serial_port` be, distinguish between the network port and the physical port.